sonarr/downloadclient.go

```diff
@@ -0,0 +1,143 @@
+	Tags                     []int                `json:"tags"`
+	Fields                   []*starr.FieldOutput `json:"fields"`
+}
+
+// GetDownloadClients returns all configured downloadclients.
+func (s *Sonarr) GetDownloadClients() ([]*DownloadClientOutput, error) {
+	return s.GetDownloadClientsContext(context.Background())
+}
+
+// GetDownloadClientsContext returns all configured downloadclients.
+func (s *Sonarr) GetDownloadClientsContext(ctx context.Context) ([]*DownloadClientOutput, error) {
+	var output []*DownloadClientOutput
+
+	req := starr.Request{URI: bpDownloadClient}
+	if err := s.GetInto(ctx, req, &output); err != nil {
+		return nil, fmt.Errorf("api.Get%s): %w", &req, err)
+	}
+
+	return output, nil
+}
+
+// GetDownloadClient returns a single downloadclient.
+func (s *Sonarr) GetDownloadClient(downloadclientID int) (*DownloadClientOutput, error) {
+	return s.GetDownloadClientContext(context.Background(), downloadclientID)
+}
+
+// GetDownloadClientContext returns a single downloadclient.
+func (s *Sonarr) GetDownloadClientContext(ctx context.Context, downloadclientID int) (*DownloadClientOutput, error) {
+	var output DownloadClientOutput
+
+	req := starr.Request{URI: path.Join(bpDownloadClient, fmt.Sprint(downloadclientID))}
+	if err := s.GetInto(ctx, req, &output); err != nil {
+		return nil, fmt.Errorf("api.Get%s): %w", &req, err)
+	}
+
+	return &output, nil
+}
+
+// AddDownloadClient creates a downloadclient.
+func (s *Sonarr) AddDownloadClient(downloadclient *DownloadClientInput) (*DownloadClientOutput, error) {
+	return s.AddDownloadClientContext(context.Background(), downloadclient)
+}
+
+// AddDownloadClientContext creates a downloadclient.
+func (s *Sonarr) AddDownloadClientContext(ctx context.Context,
+	client *DownloadClientInput,
+) (*DownloadClientOutput, error) {
+	var output DownloadClientOutput
+
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(client); err != nil {
+		return nil, fmt.Errorf("json.Marshal(%s): %w", bpDownloadClient, err)
+	}
+
+	req := starr.Request{URI: bpDownloadClient, Body: &body}
+	if err := s.PostInto(ctx, req, &output); err != nil {
+		return nil, fmt.Errorf("api.Post%s): %w", &req, err)
+	}
+
+	return &output, nil
+}
+
+// UpdateDownloadClient updates the downloadclient.
+func (s *Sonarr) UpdateDownloadClient(downloadclient *DownloadClientInput) (*DownloadClientOutput, error) {
+	return s.UpdateDownloadClientContext(context.Background(), downloadclient)
+}
+
+// UpdateDownloadClientContext updates the downloadclient.
+func (s *Sonarr) UpdateDownloadClientContext(ctx context.Context,
+	client *DownloadClientInput,
+) (*DownloadClientOutput, error) {
+	var output DownloadClientOutput
+
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(client); err != nil {
+		return nil, fmt.Errorf("json.Marshal(%s): %w", bpDownloadClient, err)
+	}
+
+	req := starr.Request{URI: path.Join(bpDownloadClient, fmt.Sprint(int(client.ID))), Body: &body}
```

Comment:
`int()` not needed.

Reply:
changed across all files, there was still something missing

Reply:
it looks like this is still here?